src/server/JasmineGraphInstanceService.cpp

```diff
@@ -80,4301 +144,5243 @@ void *instanceservicesession(void *dummyPt) {
+    char buffer[128];
+    std::string result = "SUCCESS";
+    std::string centralGraphIdentifier = graphId + +"_centralstore_" + partitionId;
+    std::string dataFolder = Utils::getJasmineGraphProperty("org.jasminegraph.server.instance.datafolder");
+    std::string aggregatorFilePath = Utils::getJasmineGraphProperty("org.jasminegraph.server.instance.aggregatefolder");
 
-            JasmineGraphInstanceService::duplicateCentralStore(serverPort, stoi(graphID), stoi(partitionID),
-                                                               workerSockets, "localhost");
-        } else if (line.compare(JasmineGraphInstanceProtocol::WORKER_IN_DEGREE_DISTRIBUTION) == 0) {
-            instance_logger.log("Received : In degree distribution to aggregator", "info");
+    if (JasmineGraphInstanceService::isInstanceCentralStoreExists(graphId, partitionId)) {
+        std::string centralStoreFile = dataFolder + "/" + centralGraphIdentifier;
+        std::string copyCommand;
 
-            write(connFd, JasmineGraphInstanceProtocol::OK.c_str(), JasmineGraphInstanceProtocol::OK.size());
-            instance_logger.log("Sent : " + JasmineGraphInstanceProtocol::OK, "info");
-            bzero(data, INSTANCE_DATA_LENGTH + 1);
-            read(connFd, data, INSTANCE_DATA_LENGTH);
-            string graphID = (data);
-            graphID = Utils::trim_copy(graphID, " \f\n\r\t\v");
-            instance_logger.log("Received Graph ID: " + graphID, "info");
+        DIR *dir = opendir(aggregatorFilePath.c_str());
+
+        if (dir) {
+            closedir(dir);
+        } else {
+            std::string createDirCommand = "mkdir -p " + aggregatorFilePath;
+            FILE *createDirInput = popen(createDirCommand.c_str(), "r");
+            pclose(createDirInput);
+        }
+
+        if (aggregatorHost == host) {
+            copyCommand = "cp " + centralStoreFile + " " + aggregatorFilePath;
+        } else {
+            copyCommand = "scp " + centralStoreFile + " " + aggregatorHost + ":" + aggregatorFilePath;
+        }
 
-            write(connFd, JasmineGraphInstanceProtocol::OK.c_str(), JasmineGraphInstanceProtocol::OK.size());
-            instance_logger.log("Sent : " + JasmineGraphInstanceProtocol::OK, "info");
+        FILE *copyInput = popen(copyCommand.c_str(), "r");
 
-            bzero(data, INSTANCE_DATA_LENGTH + 1);
-            read(connFd, data, INSTANCE_DATA_LENGTH);
-            string partitionID = (data);
-            partitionID = Utils::trim_copy(partitionID, " \f\n\r\t\v");
-            instance_logger.log("Received Partition ID: " + partitionID, "info");
+        if (copyInput) {
+            // read the input
+            while (!feof(copyInput)) {
+                if (fgets(buffer, 128, copyInput) != NULL) {
+                    result.append(buffer);
+                }
+            }
+            if (!result.empty()) {
+                std::cout << result << std::endl;
```

Comment:
We should remove this `std::cout`

Reply:
If we remove that, we do not need to read the output of the command. Therefore, we can replace `popen()` with `system()` if we remove this. Also the `buffer[128]` will become unused.

Reply:
Ok